sklearn/decomposition/tests/test_kernel_pca.py

```diff
@@ -155,6 +155,32 @@ def test_remove_zero_eig():
     assert_equal(Xt.shape, (3, 0))
 
 
+def test_leave_zero_eig():
+    """This test checks that fit().transform() returns the same result as
+    fit_transform() in case of non-removed zero eigenvalue.
```

Comment:
Is there a similar test for `remove_zero_eig=True`? should they be combined?

Reply:
Yes there was one : `test_remove_zero_eig` . Since the tests are quite different and the new one is dedicated to providing non-regression for this issue, I chose to simply insert the new one after the existing one, but not to merge them. If you feel that it makes more sense to merge them, please let me know.